test/suites/integration/grpcroute_test.go

```diff
@@ -0,0 +1,377 @@
+		Expect(err).To(BeNil())
+		Expect(*tg.Config.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*tg.Config.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionGrpc))
+		route, _ := core.NewRoute(grpcRoute)
+		latticeService = testFramework.GetVpcLatticeService(ctx, route)
+		Eventually(func(g Gomega) {
+			rules, err := testFramework.GetLatticeServiceHttpsListenerNonDefaultRules(ctx, latticeService)
+			g.Expect(err).To(BeNil())
+			g.Expect(len(rules)).To(Equal(1))
+			g.Expect(*rules[0].Match.HttpMatch.Method).To(Equal("POST"))
+			g.Expect(*rules[0].Match.HttpMatch.PathMatch.Match.Prefix).To(Equal("/"))
+		}).Should(Succeed())
+
+		fmt.Println("latticeService.DnsEntry.DomainName: ", latticeService.DnsEntry.DomainName)
+		grpcurlCmdOptions := test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             " grpcbin.GRPCBin",
+			Method:              "DummyUnary",
+			ReqParamsJsonString: `{ "f_string": "myTestString", "f_int32": 42, "f_bytes": "SGVsbG8gV29ybGQ="}`,
+			UseTLS:              true,
+		}
+
+		var stdoutStr, stderrStr string
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err = testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+		}).Should(Succeed())
+		Expect(stderrStr).To(BeEmpty())
+		Expect(stdoutStr).To(ContainSubstring(`"fString": "myTestString"`))
+		Expect(stdoutStr).To(ContainSubstring(`"fInt32": 42`))
+
+		grpcurlCmdOptions = test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             "addsvc.Add",
+			Method:              "Sum",
+			ReqParamsJsonString: `{"a": 5, "b": 6}`,
+			UseTLS:              true,
+		}
+		//Happy path: Verify client is able to invoke addsvc.Add/Sum method
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err = testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+		}).Should(Succeed())
+		Expect(err).To(BeNil())
+		Expect(stderrStr).To(BeEmpty())
+		Expect(stdoutStr).To(ContainSubstring("\"v\": " + strconv.Itoa(5+6)))
+
+		grpcurlCmdOptions = test.RunGrpcurlCmdOptions{
+			GrpcServerHostName:  *latticeService.DnsEntry.DomainName,
+			GrpcServerPort:      "443",
+			Service:             "addsvc.Add",
+			Method:              "Concat",
+			ReqParamsJsonString: `{ "a": "Str1", "b": "Str2"}`,
+			UseTLS:              true,
+		}
+		//Happy path: Verify client is able to invoke addsvc.Add/Concat method
+		Eventually(func(g Gomega) {
+			stdoutStr, stderrStr, err = testFramework.RunGrpcurlCmd(grpcurlCmdOptions)
+			g.Expect(err).To(BeNil())
+		}).Should(Succeed())
+		Expect(stderrStr).To(BeEmpty())
+		Expect(stdoutStr).To(ContainSubstring(`"v": "Str1Str2"`))
+	})
+
+	It("Update existing GRPCRoute to have only one new rule with gRPCMethod, gRPCService and header match, "+
+		"it can only invoke the matched gRPCMethod with correct headers", func() {
```

Comment:
I'm not following the second part, "it can only invoke the matched gRPCMethod with correct headers" what does it mean? What is "it", if test only for update config, do we need to test traffic? Why everything in same test case?

Reply:
"it" means client or grpcurl pod, updating grpcRoute's config means update latticeService listener rules, this change will also propagate to vpc lattice data plane. Testing traffic could give us more confident that thing can e2e work?  Trying to use few test cases to cover more controller code path, I just worry adding too many "unit test" could bloat whole e2etest suite running time

Reply:
Traffic test is a good thing, but throwing everything into test does not make it better. It's harder to read and maintain. Our gatway-api talks with lattice control plane, we should assume here that lattice CP and DP will do a right thing. Most of the test should validate that what we create from controller should be in correct state in lattice CP. 

Separation of concerns. I'm not against traffic tests, but our e2e test covers not only controller but Lattice CP and DP, and we already have extensive coverage on lattice side with canaries and other tests.